addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -584,14 +584,15 @@ var SnippetEditor = Widget.extend({
             await focusOrBlur(editor, styles);
         }
         await Promise.all(editorUIsToUpdate.map(editor => editor.updateOptionsUI()));
-        await Promise.all(editorUIsToUpdate.map(editor => editor.updateOptionsUIVisibility()));
-
-        // As the 'd-none' class is added to option sections that have no visible
-        // options with 'updateOptionsUIVisibility', if no option section is
-        // visible, we prevent the activation of the options.
-        const optionsSectionVisible = editorUIsToUpdate.some(
-             editor => !editor.$optionsSection[0].classList.contains('d-none')
-        );
+        // A `d-none` class is added to option sections that have no visible
+        // options with `updateOptionsUIVisibility`. If no option section is
+        // visible (including the options moved to the toolbar), we prevent
+        // the activation of the options.
+        const optionsSectionVisible = await Promise.all(
+            editorUIsToUpdate.map((editor) => editor.updateOptionsUIVisibility())
+        ).then(editorVisibilityValues => {
+            return editorVisibilityValues.some(editorVisibilityValue => editorVisibilityValue)
+        });
```

Comment:
> @xO-Tx Since you don't change the updateOptionsUIVisibility return, as I asked... should not this have been re-changed too? How could this work?

The return was added to `updateOptionsUIVisibility` [here](https://github.com/odoo/odoo/pull/158352/files#diff-52a4f9d2c217548e69e6b7fd097f286f1754a6389734eea254b87255e501cbefR670). But I kept it because its logic is clearer than the `updateUIVisibility`: **if no option section is visible (including the ones in the toolbar), we prevent the activation of the options...** And also the returned `optionsSectionVisible` value is the same used to toggle the `"d-non"` class that we check again in `toggleOptions()` (it was just fixed by taking toolbar options visibility into consideration).

If you think it's too risky, I can restore this code too :+1:

Reply:
@xO-Tx Yes, sorry, I changed my mind entirely: I'd really prefer a minimal diff in stable. Keep the code as much as it currently is in 17.0 and let's make it fully rely on `updateUIVisibility` return (including for the toolbar) in master only.

Reply:
Code restored to the DOM check version for stable. The other discussed updates will be added in master, just let me know if something needs to be adjusted here :+1: